src/device/pf_port.c

```diff
@@ -43,11 +43,25 @@ void pf_port_init(pnet_t * net)
    }
 }
 
-void pf_port_get_list_of_ports (pnet_t * net, pf_lldp_port_list_t * p_list)
+void pf_port_get_list_of_ports (
+   pnet_t * net,
+   uint16_t max_port,
+   pf_lldp_port_list_t * p_list)
 {
-   /* TODO: Implement support for multiple ports */
+   uint16_t port;
+   uint16_t index = 0;
+   uint8_t port_bit_mask = 0;
    memset (p_list, 0, sizeof (*p_list));
-   p_list->ports[0] = BIT (8 - 1);
+
+   CC_ASSERT (sizeof (*p_list) > (max_port / 8));
```

Comment:
To fix unittest:
`CC_ASSERT (sizeof (*p_list) >= (max_port / 8));`

Reply:
I moved the assert into the loop to avoid complex condition. 
Done